Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -851,28 +867,34 @@ class Gx2Fitter {
         // TODO: Material handling. Should be there for hole and measurement
       }
 
+      // Get required number of degrees of freedom ndfSystem
+      if (aMatrix(4, 4) == 0) {
+        ndfSystem = 4;
+      } else if (aMatrix(5, 5) == 0) {
+        ndfSystem = 5;
+      } else {
+        ndfSystem = 6;
+      }
```

Comment:
Are these all cases we have to care about? If so it might be worth commenting why and which if does what

Reply:
Should be.
`0`-`3`: We always have l0, l1, phi, theta
`4`: no magentic field -> q/p is empty
`5`: no T measurement -> T unfittable
`6`: Full fit

I will comment that.